md/src/main/java/io/codelex/streams/practice/StreamsExercise.java

```diff
@@ -0,0 +1,144 @@
+package io.codelex.streams.practice;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * TODO - search for StreamsExerciseTest and make all tests pass
+ */
+public class StreamsExercise {
+    public static List<Integer> returnSquareRoot(List<Integer> numbers) {
+        return numbers.stream().map(number -> (int) Math.sqrt(number)).toList();
+    }
+
+    public static List<Integer> getAgeFromUsers(List<User> user) {
+        return user.stream().map(User::getAge).toList();
+    }
+
+    public static List<Integer> getDistinctAges(List<User> users) {
+        return users.stream().map(User::getAge).distinct().toList();
+    }
+
+    public static List<User> getLimitedUserList(List<User> users, int limit) {
+        return users.stream().limit(limit).toList();
+    }
+
+    public static Integer countUsersOlderThen25(List<User> users) {
+        return users.stream().mapToInt(user -> user.getAge() > 25 ? 1: 0).sum();
+    }
+
+    public static List<String> mapToUpperCase(List<String> strings) {
+        return strings.stream().map(String::toUpperCase).toList();
+    }
+
+    public static Integer sum(List<Integer> integers) {
+        return integers.stream().mapToInt(integer -> integer).sum();
+    }
+
+    public static List<Integer> skip(List<Integer> integers, Integer toSkip) {
+        return integers.stream().skip(toSkip).toList();
+    }
+
+    public static List<String> getFirstNames(List<String> names) {
+        return names.stream().map(name -> Arrays.stream(name.split(" ")).toList().get(0)).toList();
+    }
+
+    public static List<String> getDistinctLetters(List<String> names) {
```

Comment:
Šeit vajadzētu mēģināt iztikt bez papildus saraksta. Te var izmantot flatMap:
names.stream().flatMap( ... )
Flat map metodei padod funkciju, kura vārdu sagriež pa burtiem un tādā veidā izveidosies stream ar String, kur katrs elements būs viens burts.
Ja izmantotu parasto map, tad katru vārdu sagriežot sanāktu, piemēram Stream<List>, bet flatMap "saplacina" un uztaisa Stream
Idejiski:
["ab", "cd", "ef"].map(s => s.split("")) būs šādi => [["a", "b"], ["c", "d"], ["e", "f"]]
Savukārt flat map:
["ab", "cd", "ef"].flatMap(s => s.split("")) būs šādi => ["a", "b", "c", "d", "e", "f"]